pkgs/by-name/ol/ollama/package.nix

```diff
@@ -141,6 +141,7 @@ goBuild (
     buildInputs =
       lib.optionals enableRocm (rocmLibs ++ [ libdrm ])
       ++ lib.optionals enableCuda [
+        cudaToolkit
```

Comment:
We had deleted the symlinkJoin on purpose and it had been tested around the time of merging. We specifically wish to avoid ever referring to the `symlinkJoin` because it inflates the closure size

Reply:
There's got to be another fix 🤔

Reply:
I have no attachment to this fix, but without this ollama was unable to find `libcudart.so.12` at runtime. 

```
Sep 07 15:59:47 sink1 ollama[1314]: /tmp/ollama1289771407/runners/cuda_v12/ollama_llama_server: error while loading shared libraries: libcudart.so.12: cannot open shared object file: No such file or directory
```

Note that you may not see this issue if you build and run on the same system, as the path will exist in the store at runtime already.

Reply:
Also note that this `cudaToolkit` is local to this file and does not come from elsewhere in nixpkgs, https://github.com/NixOS/nixpkgs/blob/996b43788bc3dfb0a772a0d666701b81dfce646b/pkgs/by-name/ol/ollama/package.nix#L82-L90

Reply:
Leaving out a necessary dependency to reduce storage usage defeats the purpose of having `ollama-cuda` in the first place. You might as well use the CPU version of `ollama` at that point.

If you can figure out how to use less space without breaking ollama when building on a separate machine or garbage collecting, I'm all for it, but correctness (keeping ollama actually working in as many situations as possible) is my primary concern.

Reply:
> Also note that this cudaToolkit is local to this file and does not come from elsewhere in nixpkgs, 

Sure this reimplements a part of `cudaPackages.cudatoolkit` from scratch, but the point remains: this only duplicates what is already listed in `buildInputs` right below

> Leaving out a necessary dependency to reduce storage usage defeats the purpose of having ollama-cuda in the first place.

I'm not suggesting to "leave out a necessary dependency", I'm suggesting that all the necessary dependencies had already been provided to the build process, and that there remains a bug in the build process that makes ollama forget some of these dependencies unless you supply the redundant symlinkJoin. This is a bug 🤷🏻